product_warranty/models/product_supplierinfo.py

```diff
@@ -29,50 +12,41 @@ class ProductSupplierInfo(models.Model):
     _inherit = "product.supplierinfo"
 
     @api.model
-    def get_warranty_return_partner(self):
-        result = [('company', 'Company'),
-                  ('supplier', 'Supplier'),
-                  ('other', 'Other'),
-                  ]
-        return result
-
-    @api.model
     def _get_default_instructions(self):
         """ Get selected lines to add to exchange """
         instruction_ids = self.env['return.instruction']\
             .search([('is_default', '=', True)], limit=1)
         return instruction_ids
 
-    @api.one
     @api.depends('warranty_return_partner')
     def _compute_warranty_return_address(self):
         """ Method to return the partner delivery address or if none, the
         default address
         """
-        return_partner = self.warranty_return_partner
-        partner_id = self.company_id.partner_id.id
-        if return_partner:
-            if return_partner == 'supplier':
-                partner_id = self.name.id
-            elif return_partner == 'company':
-                if self.company_id.crm_return_address_id:
-                    partner_id = self.company_id.\
-                        crm_return_address_id.id
-            elif return_partner == 'other':
-                if self.warranty_return_other_address:
-                    partner_id = self.\
-                        warranty_return_other_address.id
-        self.warranty_return_address = partner_id
+        for record in self:
+            return_partner = record.warranty_return_partner
+            partner_id = record.company_id.partner_id.id
+            if return_partner:
+                if return_partner == 'supplier':
+                    partner_id = record.name.id
+                elif return_partner == 'company':
+                    if record.company_id.crm_return_address_id:
+                        partner_id = record.company_id.crm_return_address_id.id
+                elif return_partner == 'other':
+                    if record.warranty_return_other_address:
+                        partner_id = record.warranty_return_other_address.id
+            record.warranty_return_address = partner_id
 
     warranty_duration = fields.Float(
         'Period',
         help="Warranty in month for this product/supplier relation. Only "
              "for company/supplier relation (purchase order) ; the  "
              "customer/company relation (sale order) always use the "
-             "product main warranty field")
+             "product main warranty field"
+    )
     warranty_return_partner = fields.Selection(
-        get_warranty_return_partner,
-        'Return type',
+        [('company', 'Company'), ('supplier', 'Supplier'), ('other', 'Other')],
```

Comment:
Why switching up the method for static definition? The method is more modular IMO